reddit/reddit_scraper.py

```diff
@@ -0,0 +1,303 @@
+    if path1.is_file():
+        # This is the absolute path we want our file to be written in
+        path2 = Path(
+            path1.resolve().parent.parent,
+            "data",
+            source,
+            current_date,
+            base_filename,
+        )
+        if not path2.is_dir():
+            path2.mkdir(parents=True)
+        # This is the file with the path that open will write to
+        path3 = Path(path2, local_filename)
+
+    # Using with to automatically close the connection when we are done with it
+    with requests.get(url, stream=True) as req:
+        # Raise an http error if there is one
+        req.raise_for_status()
+        # Write the file in binary
+        with open(path3, "wb") as video_file:
+            for chunk in req.iter_content(chunk_size=4000):
+                # If you have chunk encoded response uncomment if
+                # and set chunk_size parameter to None.
+                # if chunk:
+                video_file.write(chunk)
+
+    return local_filename
+
+
+def test_functions(download_quantity=2):
+    # Connect to Reddit's API
+    reddit = praw.Reddit("bot1", user_agent="bot1")
+    # Check that the praw.ini file worked
+    logging.info(reddit.user.me())
+    # Link to json file with streamable and reddit links as of Oct. 13, 2020
+    # https://www.reddit.com/r/PublicFreakout/top/.json?t=year
+
+    # Select the public freakout subreddit
+    sr1 = reddit.subreddit("PublicFreakout")
+    # Select the top 10 (it is selecting top 10 of the year by default)
+    top1 = sr1.top(time_filter="day", limit=download_quantity)
+    # Initialize arrays
+    urls = []
+    ### buggy1 = []
+    # Place the relevant info into the array from the selected top 3
+    for i in top1:
+        ### # Debugging error where some attribute doesn't exist
+        logging.info(len(urls))
+        ### if(len(urls) == 15):
+        ###     buggy1.append(i)
+        try:
+            urls.append(video_url_from_submission(i))
+        except:
+            print(urls.append(video_url_from_submission(i)))
+    print("urls appended, starting downloads")
+    # Download the videos
+    for i in range(len(urls)):
+        for j in range(len(urls[i])):
+            try:
+                download_video(urls[i][j], audio=True)
+            except Exception as exception:
+                # Getting this HTTP error
+                # 403 Client Error: Forbidden for url:
+                # https://v.redd.it/716d4vdxcqu41/audio?source=fallback
+                # Also, some of the posts are not videos. When printing the #
+                # length of the url, sometimes it says "errorgfycat.com" or
+                # "errori.redd.it" for example.
+                print(f"Error with post number {str(i)}")
+                print(f"Exception: {exception}")
+
+    print("reddit_scraper.py finished")
+
+
+# Take daily top 50 links from the public freakout subreddit and attempt to
+# download
+test_functions(download_quantity=2)
```

Comment:
set this back to 50, is 50 the max?

Reply:
I've [increased the download quantity](https://github.com/myHaiven/data-collection/pull/4/commits/6aaa8f67f42f7282921fa13b1bf84cdd98de9efd) back up to 50. 50 isn't the max; their [API rules](https://github.com/reddit-archive/reddit/wiki/API#rules) state 60 requests per minute via OAuth 2 as the maximum.

I think if we want more data, we can do a large initial scrape of the top videos and then use the cron to scrape the daily content.